pyfar/classes/coordinates.py

```diff
@@ -1374,16 +1374,347 @@ def show(self, mask=None, **kwargs):
+
+        if distance_measure == 'spherical':
+            # determine validate radius
+            radius = self.radius
+            delta_radius = np.max(radius) - np.min(radius)
+            if delta_radius > 1e-15:
+                raise ValueError(
+                    "find_nearest_sph only works if all points have the same \
+                    radius. Differences are larger than 1e-15")
+            radius = np.max(radius)
+
+            # convert cartesian coordinates to length on the great circle
+            distance = 2 * radius * np.arcsin(distance/(2*radius))
+
+        if self.cdim == 1:
+            if k > 1:
+                index_multi = np.moveaxis(index, -1, 0)
+                index = np.empty((k), dtype=tuple)
+                for kk in range(k):
+                    index[kk] = tuple([index_multi[kk]], )
+            else:
+                index = tuple([index], )
+        else:
+            index_array = np.arange(self.csize).reshape(self.cshape)
+            index_multi = []
+            for dim in range(self.cdim):
+                index_multi.append([])
+                for i in index.flatten():
+                    index_multi[dim].append(np.where(i == index_array)[dim][0])
+                index_multi[dim] = np.asarray(
+                    index_multi[dim]).reshape(index.shape)
+            if k > 1:
+                index_multi = np.moveaxis(index_multi, -1, 0)
+                index = np.empty((k), dtype=tuple)
+                for kk in range(k):
+                    index[kk] = tuple(index_multi[kk])
+            else:
+                index = tuple(index_multi)
+
+        if k > 1:
+            distance = np.moveaxis(distance, -1, 0)
+
+        return index, distance
+
+    def find_within(
+            self, coords, distance=0., distance_measure='euclidean',
+            atol=1e-15, return_sorted=False):
+        """
+        Find coordinates within certain distance to the query points.
+
+        Parameters
+        ----------
+        coords : pf.Coordinates
+            Coordinates to which the nearest neighbors are searched.
+        distance : number, optional
+            Maximum allowed distance to the given points ``coords``.
+            Distance must be >= 0. For just exact matches use ``0``.
+            The default is ``0``.
+        distance_measure : string, optional
+            ``'euclidean'``
+                distance is determined by the euclidean distance.
+                This is default.
+            ``'spherical'``
+                distance is determined by the great-circle distance.
+            ``'angular'``
+                distance is determined by the angles in radiant.
+        return_sorted : bool, optional
+            Sorts returned indicies if True and does not sort them if False.
+            If None, does not sort single point queries, but does sort
+            multi-point queries which was the behavior before this
+            option was added.
```

Comment:
Since we refactor, we could decide for a consistent default behavior to simplify things. I would opt for `sort=True`

Reply:
yes maybe, lets disucss it. This parameter is directly coming from the scipy.spatial.cKDTree

Reply:
we dicided to sinplify the docstring and remove the None.